lib/hoverMain.js

```diff
@@ -0,0 +1,52 @@
+//'use babel';
+//what is babel? some folks say I need it, some say I don't
+
+  // Extended: Returns the word surrounding the most recently added cursor.
+  //
+  // * `options` (optional) See {Cursor::getBeginningOfCurrentWordBufferPosition}.
+  //getWordUnderCursor (options) {
+    //return this.getTextInBufferRange(this.getLastCursor().getCurrentWordBufferRange(options))
+  //}
+
+
+
+  //This looks promising, you can hover over a regex, eg "#5" and this
+  //will catch that and show it.
+
+ import {CompositeDisposable} from 'atom';
+
+  const cursorListener = {
+  subscriptions: null,
+  cursorReturn: null,
+    activate (state) {
+      this.subscriptions = new CompositeDisposable();
+    },
+
+    //This should be the main subscription we need.
+    //Others can be added into it to get gather other events.
+    this.subscriptions.add(atom.texteditor.getWordUnderCursor()) {
```

Comment:
this should be run with the onload to connect the event

Reply:
added into main ts func

Reply:
is it something like atom.onload in the main ts?

or is it the hoverMain.onload in the main ts?

Reply:
I put it in the main ts